### PR TITLE
Refactor downsampling level dimension code

### PR DESCRIPTION
**Pull request by Philbrick**
Refactor downsampling level dimension code to function.
- Change 2/3 for cropping refactor
- No change in behavior

### CODE REVIEW

File: src/wsiToDcm.cpp [hunk 1]
```diff
@@ -129,6 +147,119 @@ int32_t get_openslide_level_for_downsample(openslide_t *osr,
...
+                                  const bool floorCorrectDownsampling,
+                                  openslide_t *osr,
+                                  const int32_t level,
+                                  const int64_t frameWidth,
+                                  const int64_t frameHeight,
+                                  SlideLevelDim *priorLevel,
+                                  DCM_Compression levelCompression,
+                                  const int64_t initialX,
+                                  const int64_t initialY,
+                                  SlideLevelDim * smallestSlideLevel) {
```
Comment (danielbeaudreau) [inline src/wsiToDcm.cpp]: smallestSlideLevel param is unused here

File: src/wsiToDcm.cpp [hunk 2]
```diff
@@ -129,6 +147,119 @@ int32_t get_openslide_level_for_downsample(openslide_t *osr,
...
+  if (preferProgressiveDownsampling) {
+    if (priorLevel != NULL) {
+      multiplicator = static_cast<double>(priorLevel->downsample);
+      downsampleOfLevel = static_cast<double>(downsample) / multiplicator;
+      // check that downsampling is going from higher to lower magnification
+      if (downsampleOfLevel >= 1.0) {
+        levelWidth = priorLevel->levelWidthDownsampled;
+        levelHeight = priorLevel->levelHeightDownsampled;
+        generateUsingOpenSlide = false;
+      }
```
Comment (danielbeaudreau) [inline src/wsiToDcm.cpp:203]: Should we be validating somewhere that downsample ints are specified in ascending order when progressive mode is enabled? Or maybe just documenting in the comment of that flag that this code path won't run unless downsamples are specified in an increasing order

**Review by danielbeaudreau — CHANGES_REQUESTED** [review]

File: src/wsiToDcm.cpp (hunk 1 above)
Comment (Philbrick) [reply src/wsiToDcm.cpp]: Removed.

**Pull request merged**